pkg/engine/jmespath/functions.go

```diff
@@ -965,3 +980,35 @@ func jpRandom(arguments []interface{}) (interface{}, error) {
 	}
 	return ans, nil
 }
+
+func jpX509Decode(arguments []interface{}) (interface{}, error) {
+	input, err := validateArg(x509_decode, arguments, 0, reflect.String)
+	if err != nil {
+		return nil, err
+	}
+	res := make(map[string]interface{})
+	p, _ := pem.Decode([]byte(input.String()))
+	if p == nil {
+		return res, errors.New("invalid certificate")
+	}
+
+	var v interface{}
+	cert, err := x509.ParseCertificate(p.Bytes)
+	if err != nil {
+		return res, errors.WithStack(err)
+	}
+	v = struct{ *x509.Certificate }{cert}
```

Comment:
Why do we need to use an embedded struct with a pointer here? Is it not possible to directly serialize it into a certificate?

Reply:
I was facing some errors related to pointers back then while trying to pass the certificate to `enc.Encode()`, but now it seems to be fine and I've switched to directly serializing it.